### PR TITLE
c-dbg-macro: A set of dbg(…) macros for C

### PR DESCRIPTION
**Pull request by eerimoq**
Add version 0.12.1.

A few macros that prints and returns the value of a given expression
for quick and dirty debugging, inspired by Rusts dbg!(…) macro and its
C++ variant.

* dbg(expr) for primitive data types (int, float, etc.), strings and pointers.

* dbgb(expr) to force boolean true/false output.

* dbga(expr, length) for array of primitive data types.

* dbgh(expr, size) for hexdump output.

* dbge(expr) for negative error codes.

* dbgbt() for a backtrace.

Just include dbg.h in your project to use it.

### CODE REVIEW

**Comment by SSE4** [comment]
`execinfo.h` seems to be Linux specific. does it support Windows at all? if not, please raise `ConanInvalidConfiguration` for unsupported platforms.

File: recipes/c-dbg-macro/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class DbgMacroConan(ConanFile):
+    name = "c-dbg-macro"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/eerimoq/dbg-macro"
+    license = "MIT"
+    description = "A dbg(...) macro for C"
+    topics = ("conan", "debugging", "macro", "pretty-printing", "header-only")
+    settings = ("compiler", )
```
Comment (SSE4) [inline recipes/c-dbg-macro/all/conanfile.py:12]:
```suggestion
    settings = ("compiler",  "build_type", "os", "arch")
```
better to specify all the settings

**Review by SSE4 — DISMISSED** [review]

File: recipes/c-dbg-macro/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "dbg-macro-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (uilianries) [inline recipes/c-dbg-macro/all/conanfile.py:27]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

File: recipes/c-dbg-macro/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
```
Comment (uilianries) [inline recipes/c-dbg-macro/all/conanfile.py:5]:
```suggestion

required_conan_version = ">=1.33.0"

```

strip_root support

Comment (uilianries) [inline recipes/c-dbg-macro/all/conanfile.py:5]:
```suggestion

required_conan_version = ">=1.33.0"

```

strip_root support

File: recipes/c-dbg-macro/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (uilianries) [inline recipes/c-dbg-macro/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

self.settings is outdated for cross-building check

Comment (uilianries) [inline recipes/c-dbg-macro/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

self.settings is outdated for cross-building check

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/c-dbg-macro/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/c-dbg-macro/all/conanfile.py:29]: bump

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**